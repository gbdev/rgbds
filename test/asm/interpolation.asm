SECTION "Test", ROM0

def NAME equs "ITEM"
def FMT equs "d"
def ZERO_NUM equ 0
def ZERO_STR equs "0"
; Defines INDEX as 100
def INDEX = 1{ZERO_STR}{{FMT}:ZERO_NUM}
; Defines ITEM_100 as "\"hundredth\""
def {NAME}_{d:INDEX} equs "\"hundredth\""
; Prints "ITEM_100 is hundredth"
PRINTLN STRCAT("{NAME}_{d:INDEX}", " is ", {NAME}_{d:INDEX})
; Purges ITEM_100
PURGE {NAME}_{d:INDEX}
ASSERT !DEF({NAME}_{d:INDEX})
