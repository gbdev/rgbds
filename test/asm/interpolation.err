error: interpolation.asm(20):
    Only numerical and string symbols can be interpolated
error: Assembly aborted (1 error)!
