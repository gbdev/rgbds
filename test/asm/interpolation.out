ITEM_100 is hundredth
foo 
